### PR TITLE
Upgrade to Wagtail 5.2.x

### PR DESCRIPTION
**Pull request by lparsons396**
**This PR upgrades Wagtail to version 5.2**

See [Monday ticket](https://torchbox.monday.com/boards/1318157635/pulses/1319732635) & [Wagtail 5.2 release notes](https://docs.wagtail.org/en/latest/releases/5.2.html#).

This branch builds on the Wagtail 5.1 upgrade https://github.com/torchbox/wagtail-torchbox/pull/544

The upgrade considerations actualise many of the deprecation warnings is the 5.1 upgrade and so few changes have been necessary here. Upgrading Wagtail has also necessitated the upgrading of django-filter (22.1 -> 23.4) and django-modelcluster (6.0 -> 6.1). All packages have been checked against the [package board](https://torchbox.monday.com/boards/1124794299) and use the appropriate tag / branch as needed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ authors = ["Torchbox Ltd"]
 [tool.poetry.dependencies]
 python = "^3.9"
 Django = "~3.2.20"
```
Comment (nickmoreton) [inline pyproject.toml:9]: Can you update this to the latest Django version please. ~3.2.23

File: pyproject.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ authors = ["Torchbox Ltd"]
...
-wagtail = "~5.0"
+wagtail = "~5.2"
 psycopg2 = "^2.9.3"
 gunicorn = {version = "^20.1.0", optional = true}
 django-pattern-library = "^1.0.0"
```
Comment (nickmoreton) [inline pyproject.toml:13]: Can you run `poetry update django-pattern-library` to bump this to 1.1.0

Comment (nickmoreton) [reply pyproject.toml:13]: Also run the same command for:

- wagtail-webstories
- wagtailmedia

**Review by nickmoreton — APPROVED** [review]
Hi @lparsons396  Thank for doing this. There's a few dependencies that could do with been bumped to latest releases before is merged.

File: pyproject.toml (hunk 1 above)
Comment (zerolab) [reply pyproject.toml:9]: `~4.2.8`, pretty please!
and if you install `django-upgrade` locally, you can run `git ls-files --others --cached --exclude-standard -- '*.py' | xargs django-upgrade --target 4.2` ;)

File: pyproject.toml [hunk 3]
```diff
@@ -7,7 +7,7 @@ authors = ["Torchbox Ltd"]
 [tool.poetry.dependencies]
 python = "^3.9"
```
Comment (zerolab) [inline pyproject.toml:8]: this would be the perfect time to make the jump to 3.11 (the Dockerfile will need updating too)

**Review by zerolab — CHANGES_REQUESTED** [review]

**Comment by lparsons396** [comment]
Thanks for your review @zerolab , let me know if there's anything else! :)

**Review by zerolab — APPROVED** [review]
:shipit:

**Comment by zerolab** [comment]
@helenb imho this is ready to go too

**Comment by helenb** [comment]
> @helenb imho this is ready to go too

From @katdom13 on slack:

"I’m also encountering this error during staging deployment for 5.2:
TypeError: cache_control didn't receive an HttpRequest. If you are decorating a classmethod, be sure to use @method_decorator.
I can replicate this. I reverted the change on staging for now and will investigate this along with the above tomorrow."

I will also need to wait to hear if 5.2 has been tested on staging.

**Comment by zerolab** [comment]
@helenb @katdom13 that would be https://github.com/wagtail/wagtail.org/pull/443/commits/3410fbca4f32cb3556bb40aff4aca07c06da29b5

File: tbx/urls.py [hunk 4]
```diff
@@ -84,7 +85,9 @@
     ),
 )
 
-Page.serve = get_default_cache_control_decorator()(Page.serve)
+Page.serve = method_decorator(get_default_cache_control_decorator(), name="serve")(
+    Page.serve
+)
```
Comment (katdom13) [inline tbx/urls.py:90]: This fixes the error on app startup
```
cache_control didn't receive an HttpRequest. If you are decorating a classmethod, be sure to use @method_decorator.
```

image

**Comment by katdom13** [comment]
> > @helenb imho this is ready to go too
> 
> From @katdom13 on slack:
> 
> "I’m also encountering this error during staging deployment for 5.2: TypeError: cache_control didn't receive an HttpRequest. If you are decorating a classmethod, be sure to use @method_decorator. I can replicate this. I reverted the change on staging for now and will investigate this along with the above tomorrow."
> 
> I will also need to wait to hear if 5.2 has been tested on staging.

Hi @helenb ,
Apologies, I reverted this yesterday on staging when I saw the errors, but failed to remove the tag here in this MR. Staging should still be in 5.1

**Comment by katdom13** [comment]
Hi @zerolab ,


> > `TypeError: cache_control didn't receive an HttpRequest. If you are decorating a classmethod, be sure to use @method_decorator.`
> > @helenb @katdom13 the fix for that would be [wagtail/wagtail.org@3410fbc](https://github.com/wagtail/wagtail.org/commit/3410fbca4f32cb3556bb40aff4aca07c06da29b5)

I saw this just now and updated the fix accordingly. Could you please review the changes starting at:
https://github.com/torchbox/wagtail-torchbox/pull/558/commits/bf060556c1a0267ff00bf1c896f86900eac548d5

Thanks!

**Comment by zerolab** [comment]
@katdom13 they LGTM. Added another fix (e7d3bb5)

**Comment by helenb** [comment]
@katdom13 I've created this ticket for Anna to test this on staging so we can be clear when this is ready to be merged: https://torchbox.monday.com/boards/1192293412/pulses/1347671612

**Pull request merged**